win32ss/user/user32/misc/imm.c

```diff
@@ -3,46 +3,35 @@
  * PROJECT:         ReactOS user32.dll
  * FILE:            win32ss/user/user32/misc/imm.c
  * PURPOSE:         User32.dll Imm functions
- * PROGRAMMER:      Dmitry Chapyshev (dmitry@reactos.org)
+ * PROGRAMMERS:     Dmitry Chapyshev (dmitry@reactos.org)
+ *                  Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
  * UPDATE HISTORY:
  *      01/27/2009  Created
  */
 
 #include <user32.h>
-
-#include <winnls32.h>
 #include <strsafe.h>
 
 WINE_DEFAULT_DEBUG_CHANNEL(user32);
 
 #define IMM_INIT_MAGIC 0x19650412
 
-
-Imm32ApiTable gImmApiEntries = {0};
 HINSTANCE ghImm32 = NULL;
 BOOL bImmInitializing = FALSE;
-BOOL ImmApiTableZero = TRUE;
 
+/* define stub functions */
+#undef DEFINE_IMM_ENTRY
+#define DEFINE_IMM_ENTRY(type, name, params, retval) \
+    static type WINAPI IMMSTUB_##name params { return (type)retval; }
```

Comment:
This line defines the stubs. If void, then unable to return.

Reply:
I see...
I will try to see if I could find a nice way to overcome the problem.

Reply:
```c
#define IMM_RETURN_VOID(retval) /* empty */
#define IMM_RETURN_NONVOID(retval) return retval
...
IMM_RETURN_##rettype
```

Reply:
That could work, indeed :D 👍

Reply:
OK, Done.